frontend/test/pytest/test_transform.py

```diff
@@ -121,44 +166,896 @@ def qfunc(data, x, weights):
-                qml.expval(qml.PauliZ(0) @ qml.PauliZ(1)),
-                qml.expval(qml.PauliX(0)),
-                qml.expval(qml.PauliZ(1)),
-                qml.expval(qml.PauliX(1) @ qml.PauliX(4)),
-                qml.expval(qml.PauliX(3)),
-                qml.expval(qml.PauliY(5)),
-            )
+            qml.S(0)
+            qml.RX(0.1234, 0)
+            return qml.expval(qml.X(0))
 
         return qfunc
 
     qnode_control = qnode_builder("default.qubit")
     qnode_backend = qnode_builder(backend)
-    expected = jax.jit(qnode_control)()
-    observed = qjit(qnode_backend)()
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="cond does not work with qml.RY")
+def test_defer_measurements(backend):
+    """Test defer_measurements"""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.defer_measurements
+        @qml.qnode(qml.device(device_name, wires=2), interface="jax")
+        def qfunc():
+            qml.RY(0.123, wires=0)
+            qml.Hadamard(wires=1)
+            m_0 = qml.measure(1)
+            qml.cond(m_0, qml.RY)(np.pi / 4, wires=0)
+            return qml.expval(qml.Z(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    expected = jax_jit()
+    observed = compiled()
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+def test_diagonalize_measurements(backend):
+    """Test diagonalize_measurements."""
+
+    def qnode_builder(device_name):
+        """Builder"""
+
+        @qml.transforms.diagonalize_measurements
+        @qml.qnode(qml.device(device_name, wires=3), interface="jax")
+        def qfunc(x):
+            qml.RY(x[0], wires=0)
+            qml.RX(x[1], wires=1)
+            return qml.expval(qml.X(0) @ qml.Z(1)), qml.var(0.5 * qml.Y(2) + qml.X(0))
+
+        return qfunc
+
+    qnode_control = qnode_builder("default.qubit")
+    qnode_backend = qnode_builder(backend)
+
+    jax_jit = jax.jit(qnode_control)
+    compiled = qjit(qnode_backend)
+
+    x = [np.pi / 4, np.pi / 4]
+    expected = jax_jit(x)
+    observed = compiled(x)
+    _, expected_shape = jax.tree_util.tree_flatten(expected)
+    _, observed_shape = jax.tree_util.tree_flatten(observed)
+
+    assert np.allclose(expected, observed)
+    assert expected_shape == observed_shape
+
+
+@pytest.mark.xfail(reason="cond does not work with qml.RY")
```

Comment:
Will this work with applying `qml.RY` via a function, i.e., using callable for `true` branch? 🤔

Reply:
I guess #449 really needs to be fixed! I will take a look,

Reply:
The fix is likely non-trivial, so no need to wait for it.

Reply:
#1232